test/jdk/java/awt/Focus/RequestFocusOwnerTest/RequestFocusOwnerTest.java

```diff
@@ -0,0 +1,236 @@
+    private volatile static Dimension buttonSize;
+    private volatile static boolean focusGained = false;
+    private volatile static boolean requestStatus;
+
+    private volatile static FocusListener listener = new FocusListener() {
+
+        @Override
+        public void focusLost(FocusEvent e) {
+            System.out.println(e.getSource() + ": lost focus.");
+        }
+
+        @Override
+        public void focusGained(FocusEvent e) {
+            System.out.println(e.getSource() + ": gained focus.");
+            focusGained = true;
+        }
+    };
+
+    private static void initializeGUI() {
+        frame = new Frame("Test Frame");
+        frame.setLayout(new FlowLayout());
+        button = new Button("Button");
+        button.addFocusListener(listener);
+        frame.add(button);
+        textField = new TextField(15);
+        textField.addFocusListener(listener);
+        frame.add(textField);
+        choice = new Choice();
+        choice.addItem("One");
+        choice.addItem("Two");
+        choice.addFocusListener(listener);
+        choice.setEnabled(false);
+        frame.add(choice);
+        checkbox = new Checkbox("Checkbox");
+        checkbox.addFocusListener(listener);
+        frame.add(checkbox);
+        list = new List();
+        list.add("One");
+        list.add("Two");
+        list.add("Three");
+        list.add("Four");
+        list.add("Five");
+        list.addFocusListener(listener);
+        list.setFocusable(false);
+        frame.add(list);
+        frame.pack();
+        frame.setLocationRelativeTo(null);
+        frame.setVisible(true);
+    }
+
+    public static void main(String[] args)
+        throws InvocationTargetException, InterruptedException, AWTException {
+        try {
+
+            EventQueue.invokeAndWait(RequestFocusOwnerTest::initializeGUI);
+
+            Robot robot = new Robot();
+            robot.setAutoDelay(500);
+            robot.setAutoWaitForIdle(true);
+            EventQueue.invokeAndWait(() -> {
+                buttonAt = button.getLocationOnScreen();
+                buttonSize = button.getSize();
+            });
+
+            if (!button.hasFocus()) {
+                focusGained = false;
+
+                robot.mouseMove(buttonAt.x + buttonSize.width / 2,
+                    buttonAt.y + buttonSize.height / 2);
+                robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
+                robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
+
+                robot.waitForIdle();
+
+                if (!focusGained) {
+                    throw new RuntimeException(
+                        "FAIL: Button did not gain focus when clicked. Test cannot proceed!");
+                }
+            }
+
+            focusGained = false;
+            EventQueue.invokeAndWait(() -> {
+                requestStatus = checkbox.requestFocusInWindow();
+            });
+
+            robot.waitForIdle();
+
+            if (!requestStatus) {
```

Comment:
According to the requestFocusInWindow() documentation https://docs.oracle.com/en/java/javase/20/docs/api/java.desktop/java/awt/Component.html#requestFocusInWindow() . Developers must never assume that this Component is the focus owner until this Component receives a FOCUS_GAINED event.   

So dn't depend on requestFocusInWindow()  return value